core/unix/os.c

```diff
@@ -7444,26 +7445,37 @@ pre_system_call(dcontext_t *dcontext)
             size_t sizemask;
         } data_t;
         data_t *data = (data_t *)sys_param(dcontext, 5);
-        kernel_sigset_t *mask = (kernel_sigset_t *)data->sigmask;
-        size_t sizemask = (size_t)data->sizemask;
-        kernel_sigset_t nullsigmask;
-        memset(&nullsigmask, 0, sizeof(nullsigmask));
-        /* syscall ABI, struct is in the  app's stack */
-        safe_write_ex((void *)&data->sigmask, sizeof(kernel_sigset_t),
-                      (const void *)&nullsigmask, NULL);
+        kernel_sigset_t *mask;
+        if (!safe_read(&data->sigmask, sizeof(kernel_sigset_t), &mask)) {
```

Comment:
bug: wrong sizeof!  This is a read overflow on 32-bit.

In general, always use sizeof the actual field being written to, instead of a type, to avoid bugs where the field's type changes or bugs like this one where the wrong type is there: so `sizeof(mask)`.

Reply:
Thanks